src/main/scala/dpla/ingestion3/EnricherMain.scala

```diff
@@ -0,0 +1,124 @@
+package dpla.ingestion3
+
+import java.io._
+
+import com.databricks.spark.avro._
+import org.apache.avro.Schema
+import org.apache.log4j.LogManager
+import org.apache.spark.SparkConf
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{Row, SparkSession}
+import org.apache.spark.storage.StorageLevel
+import org.eclipse.rdf4j.model.Model
+import org.eclipse.rdf4j.model.util.ModelBuilder
+import org.eclipse.rdf4j.rio._
+import dpla.ingestion3.utils.Utils
+import dpla.ingestion3.enrichments.Spatial
+
+/**
+ * Driver for running an enrichment
+ *
+ * This is a prototype that is hardcoded to run the spatial enrichment,
+ * SpatialEnricher.
+ *
+ * See https://digitalpubliclibraryofamerica.atlassian.net/wiki/display/TECH/Ingestion+3+Storage+Specification#Ingestion3StorageSpecification-S3BucketFolderStructureandContents
+ * with suggestions for how to organize input and output locations. 
+ *
+ * Example invocation:
+ * $ sbt "run-main dpla.ingestion3.EnricherMain local[2] <from> <to>"
+ * ... where <from> is the Avro resource (file / directory) and <to> is the
+ *     destination (directory or s3 folder)
+ *
+ */
+object EnricherMain {
+
+  // Avro schema for our output
+  val schemaStr: String =
+    """
+    {
+      "namespace": "la.dp.avro",
+      "type": "record",
+      "name": "MAPRecord.v1",
+      "doc": "Prototype enriched record",
+      "fields": [
+        {"name": "id", "type": "string"},
+        {"name": "document", "type": "string"}
+      ]
+    }
+    """
+  val logger = LogManager.getLogger(EnricherMain.getClass)
+
+  /*
+   * args:
+   *   master: String -- Spark master descriptor
+   *   inputURI: String -- URI of Avro input
+   *   outputURI: String -- URI of Avro output
+   */
+  def main(args: Array[String]) {
+
+    if (args.length != 3) {
+      logger.error("Arguments should be <master>, <inputURI>, <outputURI>")
+      sys.exit(-1)
+    }
+    val sparkMaster = args(0)
+    val inputURI = args(1)
+    val outputURI = args(2)
+
+    val start_time = System.currentTimeMillis()
+
+    val sparkConf = new SparkConf()
+      .setAppName("Enricher")
+      .setMaster(sparkMaster)
+    val spark = SparkSession.builder().config(sparkConf).getOrCreate()
+    val sc = spark.sparkContext
+
+    val rdd: RDD[Row] = spark.read
+      .format("com.databricks.spark.avro")
+      .option("avroSchema", schemaStr)
+      .load(inputURI)
+      .rdd
+
+    // Process all of the records and wrap them with a dataframe
+    val rows = rdd.map(
+      record => (
+        record.getAs[String]("id"),
+        process(record.getAs[String]("document"))
+      )
+    )
+    val dataframe = spark.createDataFrame(rows)
```

Comment:
Probably a better thing to do here is to call `toDF("id", "document")` on the `RDD`. This is probably the only feedback I have that you should probably implement right away, because right now your resulting DataFrame doesn't have the right column labels and therefore doesn't actually conform to the applied Avro schema.